fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/DimensionRenderingRegistry.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendering.v1;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.client.render.DimensionEffects;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.RegistryKey;
+import net.minecraft.world.World;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.client.rendering.DimensionRenderingRegistryImpl;
+
+/**
+ * Dimensional renderers render world specific visuals of a world.
+ * They may be used to render the sky, weather, or clouds.
+ * The {@link DimensionEffects} is the vanilla dimensional renderer.
+ */
+@Environment(EnvType.CLIENT)
+public interface DimensionRenderingRegistry {
+	/**
+	 * Registers the custom sky renderer for a {@link World}.
+	 *
+	 * <p>This overrides Vanilla's sky rendering.
+	 * @param key A {@link RegistryKey} for your {@link World}
+	 * @param renderer A {@link SkyRenderer} implementation
+	 * @throws IllegalArgumentException if key is already registered.
```

Comment:
These IAEs aren't thrown anymore so this is outdated (also applies to the other registration methods)